### PR TITLE
FSI: Implement arrow dictionary importing

### PR DESCRIPTION
**Pull request by fexolm**
It's just a reference implementation. More optimizations and refactoring will come during the global FSI refactoring.

### CODE REVIEW

**Review by anton-malakhov — CHANGES_REQUESTED** [review]
please add/modify a test

File: DataMgr/ForeignStorage/ArrowForeignStorage.cpp [hunk 1]
```diff
@@ -625,6 +645,79 @@ void ArrowForeignStorageBase::createDictionaryEncodedColumn(
...
+void ArrowForeignStorageBase::convertArrowDictionary(
+    StringDictionary* dict,
+    const ColumnDescriptor& c,
+    std::vector<ArrowFragment>& col,
+    arrow::ChunkedArray* arr_col_chunked_array,
+    tbb::task_group& tg,
+    const std::vector<Frag>& fragments,
+    ChunkKey key,
+    Data_Namespace::AbstractBufferMgr* mgr) {
+  tg.run([this, dict, &c, &col, arr_col_chunked_array, &tg, &fragments, k = key, mgr]() {
```
[the remark this answers is not in the document]
Comment (fexolm) [reply DataMgr/ForeignStorage/ArrowForeignStorage.cpp:657]: I hope to refactor all FSI one day, so, I'll do that later.

**Comment by fexolm** [comment]
> please add/modify a test

We don't have tests for embedded FSI yet, it would take some time to add them. Let's wait with them until I could spend all my time to FSI.

**Comment by fexolm** [comment]
I'll close this and continue working on it in https://github.com/intel-ai/omniscidb/pull/81

**Pull request closed without merging**